src/components/ScaleControls.vue

```diff
@@ -53,13 +55,24 @@ defineExpose({ focus, clearPaletteInfo })
       <input
         id="base-midi-note"
         type="number"
-        step="1"
+        :step="stepx"
         v-model="scale.baseMidiNote"
         @input="updateScale()"
       />
       <span class="midi-name">{{
-        midiNoteNumberToName(scale.baseMidiNote, -1, scale.accidentalPreference)
-      }}</span>
+          midiNoteNumberToName(scale.baseMidiNote, -1, scale.accidentalPreference)
+        }}</span>
+
+      <span>
```

Comment:
It's been a while since I've programmed Vue, but it's something along the lines of adding the new preference, `const state = useStateStore()` and replacing this line with:  `<span v-if="state.showMidiStepVariants">`.